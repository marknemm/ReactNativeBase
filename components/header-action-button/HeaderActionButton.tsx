import { Button, ButtonProps } from '@rneui/themed';
import { forwardRef } from 'react';
import { useStyles } from './HeaderActionButton.styles';

/**
 * The {@link HeaderActionButton} component.
 *
 * @param props The component {@link ButtonProps}.
 * @param ref The component reference.
 * @returns The {@link HeaderActionButton} component.
 */
const HeaderActionButton: React.FC<ButtonProps> = forwardRef((props, ref) => {
  const styles = useStyles(props);

  return (
    <Button
      type="clear"
      {...props}
      ref={ref}
      titleStyle={styles.title}
    />
  );
});

export default HeaderActionButton;
