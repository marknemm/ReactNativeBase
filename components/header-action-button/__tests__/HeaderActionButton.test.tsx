import HeaderActionButton from '@components/header-action-button/HeaderActionButton';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen } from '@testing-library/react-native';
import { StyleProp, TextStyle } from 'react-native';

describe('<HeaderActionButton />', () => {
  const title = 'test';

  describe('styles', () => {
    it('correctly applies default title style', () => {
      render(
        <HeaderActionButton
          title={title}
          titleStyle={{ fontWeight: 'bold' }}
        />,
        { wrapper: AppProvider }
      );

      const buttonTitle = screen.getByText(title);
      expect(buttonTitle).toHaveStyle({ color: 'white' });
    });

    it('correctly applies custom title styles', () => {
      const titleStyle: StyleProp<TextStyle> = {
        color: 'black',
        fontWeight: 'bold',
      };

      render(
        <HeaderActionButton
          title={title}
          titleStyle={titleStyle}
        />,
        { wrapper: AppProvider }
      );

      const buttonTitle = screen.getByText(title);
      expect(buttonTitle).toHaveStyle(titleStyle);
    });
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <HeaderActionButton title={title} />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
