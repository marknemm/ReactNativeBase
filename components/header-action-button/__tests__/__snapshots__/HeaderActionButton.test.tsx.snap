// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderActionButton /> snapshots renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "borderRadius": 3.142857142857143,
      },
      undefined,
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#007aff",
          "borderRadius": 3.142857142857143,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12.571428571428571,
          "paddingHorizontal": 14.571428571428571,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;
